### ISSUE TITLE
Does cdule  support the following capabilities ?

### ISSUE THREAD

**Issue by bhushanibm** [deepaksinghvi/cdule#2]
1).Lets say if we register some job whose time is elapsed, does it trigger those jobs whose time have elapsed or it will only trigger for next iteration ? If this library is used in Kubernetes Env, it would get packaged as part of Docker pod. There might be case where the pod might be down might be for some patch upgrade. During those scenario when the pod comes back, will it trigger those jobs which were scheduled when pod was down ? This is more on clarification for my earlier question no 2.

2) Does it support job which are scheduled in timezone which has daylight scheme in to effect ? When I say timezone support, I mean let say the kubernetes env which is running in asia pacific timezone. Does it support recurring daily jobs to run with US timezone and also support daylight for those job where in library modifies the trigger time based on its effect ?
3)Lets says if cdule is used in Kubernetes and is packaged along with pod. If scaling of pod happens, does job gets triggered multiple times ?
4) Is there any limit on number of concurrent jobs which can be triggerred?

**Comment by bhushanibm** [comment]
@deepaksinghvi :- Can you please respond to above queries if we are still actively pursuing the library ?

**Comment by deepaksinghvi** [comment]
Hello @bhushanibm thanks for your queries.

1. If a job is elapsed, only the next cycle jobs will be picked and triggered. If we pod comes back it gets registered and from that time onwards it will pick the job if they are eligible to be executed.
2. No it works on the UTC, so anytime it gets added it will be converted and will execute at that time.
3. About the limit it's not been evaluated yet. Some of the open items are mentioned here. https://github.com/users/deepaksinghvi/projects/1

Please do let me know if there is anything you need that is a blocker so I can try to add it quickly.

**Comment by deepaksinghvi** [comment]
@bhushanibm please check the demo here https://www.youtube.com/embed/lsgUQdU9CX4 and more details in the wiki https://github.com/deepaksinghvi/cdule/wiki

**Comment by bhushanibm** [comment]
1. Do we support execute only once type of job ? 
2. Does the above demo demonstrates that we can achieve distributed scheduling where in the job will be triggered in one of the pod and it will not be triggerred in other pods ? If so how it is achieved ?
3. We are also looking for support to nosql database such as mongodb is there a plan to support it ?

**Comment by bhushanibm** [comment]
@deepaksinghvi :- Can you please respond to above queries when you have some time?

**Comment by deepaksinghvi** [comment]
1. Yes only once by a pod out of all for a job, the user can add multiple jobs as you can see in the example.
2. Yes it takes care of the distributed scheduling.
3. Currently NoSQL is not supported but should be easy to add it. If you want to add support for it then please go ahead in a similar way as the other user has added support for MySQL https://github.com/deepaksinghvi/cdule/pull/1 else I will plan to add it later.

**Comment by apuatcfbd** [comment]
Hello @deepaksinghvi thank you for above information. I just wondering if it is possible to schedule job for once with delay. Like: schedule _`jobA` to be executed 30 minutes later_. or by a date like _`jobB` for 2024-11-23 16:18:36_ ?
These jobs are for onetime execution, no recurring wanted.

**Comment by deepaksinghvi** [comment]
> Hello @deepaksinghvi thank you for above information. I just wondering if it is possible to schedule job for once with delay. Like: schedule _`jobA` to be executed 30 minutes later_. or by a date like _`jobB` for 2024-11-23 16:18:36_ ? These jobs are for onetime execution, no recurring wanted.

Yes @apuatcfbd that would be possible, its a matter of writing the cron expression.
